### ISSUE TITLE
Job history / job stats bug

### ISSUE THREAD

**Issue by DavidWiseman** [trimble-oss/dba-dash#712]
When the DBA Dash service first starts it collects the job history available on the server from up to the last 7 days.  How much is actually collected depends on the retention settings.  After that, the service will only collect what is new since the last collection.

The potential issue is when the retention is set to less than 7 days.  If the service is restarted there is a possibility for older data to get re-imported.  Apart from the retention settings not working correctly, the JobStats_60MIN table will get updated with history that was imported previously - leading to some double counting.  

Options:
* The 7 days of backfill for the job history might be overkill - reducing this would make the problem less likely to occur.  It's useful to have when first setting up DBA Dash though.  It might be worth having this configurable regardless.  It's possible that collecting 7 days of history could be a burden on some systems.
* We could check the retention policy and ensure we don't import data outside of the retention.
* We could check the last job instance_id inserted for the SQL Instance and use this to only insert new history.

**Comment by DavidWiseman** [comment]
Note: Although we have an index on InstanceID and instance_id the index is partitioned so we can't use it to efficiently get the max instance_id for a given InstanceID.  This option is more efficient:

```
SELECT TOP(1) m.instance_id
FROM sys.partitions p
OUTER APPLY(SELECT MAX(instance_id) AS instance_id
			FROM dbo.JobHistory 
			WHERE InstanceID = @InstanceID
			AND $PARTITION.PF_JobHistory(RunDateTime) = p.partition_number
			) m
WHERE p.object_id = OBJECT_ID('dbo.JobHistory')
AND p.index_id=1
AND p.rows > 0
AND m.instance_id IS NOT NULL
ORDER BY p.partition_number DESC
```

**Comment by DavidWiseman** [comment]
Fixed in 2.48.0
